ripngd/ripng_interface.c

```diff
@@ -184,7 +184,8 @@ static int ripng_if_down(struct interface *ifp)
 			zlog_debug("turn off %s", ifp->name);
 
 		/* Leave from multicast group. */
-		ripng_multicast_leave(ifp, ripng->sock);
+		if (ripng)
```

Comment:
This fixes the warning, but I guess (depending on the implementation) an `assert` would be more correct: assuming `ripng` could be set to NULL and have joined multicast group previously, this would mean that we would have never left that group (e.g. hanging multicast group forever).

If no ripngd experts say otherwise I think this is good enough and that situation shouldn't happen.